persistence/nosql/async/api/src/main/java/org/apache/polaris/async/AsyncExec.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.async;
+
+import static java.util.concurrent.Executors.callable;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+
+/**
+ * Abstraction for platform/environment-specific scheduler implementations.
+ *
+ * <p>Quarkus production systems use Vert.x, tests usually use Java executors.
```

Comment:
Even if not using Java executors, can't we still use Java *interfaces* like `ScheduledExecutorService`? Is Vertx fundamentally incompatible with ScheduledFutures that are returned in `ScheduledExecutorService`?